data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/utils/UtilHelper.java

```diff
@@ -0,0 +1,94 @@
+package gov.cdc.etldatapipeline.organization.utils;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import gov.cdc.etldatapipeline.organization.model.DataRequiredFields;
+import gov.cdc.etldatapipeline.organization.model.avro.DataEnvelope;
+import gov.cdc.etldatapipeline.organization.model.avro.DataField;
+import gov.cdc.etldatapipeline.organization.model.avro.DataSchema;
+import gov.cdc.etldatapipeline.organization.model.odse.DebeziumMetadata;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.reflect.Field;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+@Slf4j
+@NoArgsConstructor
+public class UtilHelper {
+    private final ObjectMapper objectMapper = new ObjectMapper()
+            .registerModule(new JavaTimeModule());
+
+    private static UtilHelper utilHelper;
+
+    public static UtilHelper getInstance() {
+        utilHelper = (utilHelper == null) ? new UtilHelper() : utilHelper;
+        return utilHelper;
+    }
+
+    public <T> T deserializePayload(String jsonString, Class<T> type) {
+        try {
+            if (jsonString == null) return null;
+            return objectMapper.readValue(jsonString, type);
+        } catch (JsonProcessingException e) {
+            log.error("JsonProcessingException: ", e);
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public <T extends DebeziumMetadata> T deserializePayload(
+            String jsonString, String nodeName, Class<T> type) {
+        try {
+            JsonNode jsonTree = objectMapper.readTree(jsonString);
+            JsonNode rootNode = jsonTree.at(nodeName);
+            T dMetadata = objectMapper.convertValue(rootNode, type);
+            if (!Objects.isNull(dMetadata)) {
+                if (!Objects.isNull(jsonTree.at("/payload/ts_ms")))
+                    dMetadata.setTs_ms(jsonTree.at("/payload/ts_ms").asLong());
+                if (!Objects.isNull(jsonTree.at("/payload/op")))
+                    dMetadata.setOp(jsonTree.at("/payload/op").asText());
+            }
+            return dMetadata;
+        } catch (JsonProcessingException e) {
+            log.error("JsonProcessingException: ", e);
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public <T extends DataRequiredFields> DataEnvelope buildAvroRecord(T obj) {
```

Comment:
Make sure using generics is not increasing the complexity or affecting any performance overhead in the code.

Reply:
sure. we might need this generic here for the schema construction.